libp2p/timed_cache/basic_time_cache.py

```diff
@@ -0,0 +1,51 @@
+import threading
+import time
+
+
+class TimedCache:
```

Comment:
This would be a good place to use `ABC` like:

```python
from abc import ABC

class BaseTimedCache(ABC)
```

You can then decorate the `add` and `has` methods with `@abstractmethod` and you don't have to worry about raising anything, it's handled by the `ABC`.